tests/integration_tests/modules/test_ssh_keysfile.py

```diff
@@ -83,3 +73,103 @@ def test_authorized_keys(client: IntegrationInstance):
                     look_for_keys=False,
                     allow_agent=False,
                 )
+
+        # Also ensure we haven't messed with any /home permissions
+        # See LP: #1940233
+        home_dir = '/home/{}'.format(user)
+        home_perms = '755'
+        if user == 'root':
+            home_dir = '/root'
+            home_perms = '700'
+        assert '{} {}'.format(user, home_perms) == client.execute(
+            'stat -c "%U %a" {}'.format(home_dir)
+        )
+        if client.execute("test -f {}/.ssh".format(home_dir)).ok:
```

Comment:
Does this actually work? I get exit 1  from this call meaning we don't drop in for the perms check.

Reply:
Also can we attempt to run something like the example ssh-keygen call in the bug report to confirm that all the plumbing works as expected without error?